src/Controller/Component/PrgComponent.php

```diff
@@ -53,6 +54,40 @@ public function startup()
         return $this->_registry->getController()->redirect($url);
     }
 
+    /**
+     * Populates the $_isSearch view variable based on the current request.
+     *
+     * You need to configure the modelClass config if you are not using the controller's
+     * default modelClass property.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function beforeRender()
+    {
+        if (!$this->_actionCheck()) {
```

Comment:
I realized we should also bail out if response type is not `text/html`, else setting the view var could inadvertently modify API response.

Reply:
Fair point.

Reply:
Actually, thinking about it some more: Wouldn't this modify the response in a good way, just adding the additional info here?
I mean, if one involves the Prg component, then probably not for a stateless, but somewhat mixed or stateful API, and here one would probably also want to act on the info on whether the request contains a search filter or not.
What's the harm here in letting it add this on top by default?

Reply:
The same controller could be used for html as well as json responses. Harmful or not unexpected/unwanted changes in output is a bad thing.

Reply:
The same controller could be used for html as well as json responses. Harmful or not unexpected/unwanted changes in output is a bad thing.